app/src/org/commcare/android/database/MigrationException.java

```diff
@@ -0,0 +1,28 @@
+package org.commcare.android.database;
+
+/**
+ * Exception thrown when an error is encountered during global db upgrades
+ *
+ * @author amstone
+ */
+public class MigrationException extends RuntimeException {
+
+    public static final String DEFINITE_FAILURE_MESSAGE = "CommCare was unable to to migrate your " +
```

Comment:
Since these aren't semantically meaningful, I would move the messages themselves to the Android Strings resource so they can be localized by the Android system if we needed